moto/sqs/models.py

```diff
@@ -1004,6 +1006,12 @@ def untag_queue(self, queue_name, tag_keys):
     def list_queue_tags(self, queue_name):
         return self.get_queue(queue_name)
 
+    def is_message_valid_based_on_retenention_period(self, queue_name):
+        message_attributes = self.get_queue_attributes(queue_name, [])
+        if message_attributes.get("MessageRetentionPeriod") + message_attributes.get("CreatedTimestamp") <= unix_time():
```

Comment:
This is probably more readable by introducing a local variable:
`if retain_until <= unix_time()`